src/Compilers/CSharp/Portable/Parser/SlidingTextWindow.cs

```diff
@@ -333,6 +333,34 @@ public char PeekChar(int delta)
             return ch;
         }
 
+        public bool NextAre(string chars, int offset = 0)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(chars));
+            var n = chars.Length;
+            if (!CanGet(offset + n - 1)) return false;
+            for (var i = 0; i < n; i++)
+            {
+                if (chars[i] != PeekChar(offset + i)) return false;
+            }
+            return true;
```

Comment:
this looks like this could be much more easily written just using PeekChar

Reply:
The implementation is the same as the VB one. VB doesn't return an "InvalidCharacter" when out of range.